be/src/cloud/cloud_tablet.cpp

```diff
@@ -267,6 +282,248 @@ TabletSchemaSPtr CloudTablet::merged_tablet_schema() const {
     return _merged_tablet_schema;
 }
 
+bool CloudTablet::split_rowsets_by_version_overlap(
+        const std::vector<RowsetSharedPtr>& input_rowsets,
+        std::vector<RowsetSharedPtr>* new_rowsets,
+        std::vector<RowsetSharedPtr>* overlapping_rowsets) {
+    auto max_version = max_version_unlocked();
+    for (auto rs : input_rowsets) {
+        if (rs->version().first > max_version) {
+            new_rowsets->push_back(rs);
+        } else if (rs->version().second <= max_version) {
+            overlapping_rowsets->push_back(rs);
+        } else {
```

Comment:
The else branch is not possile?

Reply:
e.g. max_version = 20, we've synced rowset [15-25][26-30][31-31]
[15-25] is in the else branch
I'll add such case into the comment